handler/core/src/main/java/io/knotx/fragments/handler/consumer/FragmentHtmlBodyWriterFactory.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (C) 2019 Knot.x Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.knotx.fragments.handler.consumer;
+
+import io.knotx.fragments.api.Fragment;
+import io.knotx.fragments.engine.FragmentEvent;
+import io.knotx.server.api.context.ClientRequest;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+public class FragmentHtmlBodyWriterFactory implements FragmentEventsConsumerFactory {
+
+  static final String FRAGMENT_TYPES_OPTIONS = "fragmentTypes";
+  static final String CONDITION_OPTION = "condition";
+  static final String HEADER_OPTION = "header";
+  static final String PARAM_OPTION = "param";
+
+  @Override
+  public String getName() {
+    return "fragmentHtmlBodyWriter";
+  }
+
+  @Override
+  public FragmentEventsConsumer create(JsonObject config) {
+    return new FragmentEventsConsumer() {
+      private Set<String> supportedTypes = getSupportedTypes(config);
+      private String requestHeader = getConditionHeader(config);
+      private String requestParam = getConditionParam(config);
+
+      @Override
+      public void accept(ClientRequest request, List<FragmentEvent> events) {
+        if (containsHeader(request) || containsParam(request)) {
```

Comment:
Is that enough to check if header/param is present, not checking their value?

Reply:
I think it is not required in this case. We can consider it adding some security improvements.

Reply:
I think it is enough in this case. We can consider some "security" improvement in the future.